### PR TITLE
adding timeout for flaky tests

### PR DESCRIPTION
**Pull request by se468**
can override default timeout

### CODE REVIEW

File: src/xdotoolify.js [hunk 1]
```diff
@@ -394,7 +436,7 @@ _Xdotoolify.prototype._do = async function(command) {
   }
   if (command) {
     //console.log('command is ' + command);
-    childProcess.execSync('xdotool ' + command);
+    return await childProcess.execSync('xdotool ' + command);
```
Comment (stoarca) [inline src/xdotoolify.js]: why do you need await here? execSync is synchronous

File: src/xdotoolify.js [hunk 2]
```diff
@@ -298,10 +331,17 @@ _Xdotoolify.prototype.do = async function() {
...
+
         var pos = op.selector;
         if (typeof op.selector === 'string' || Array.isArray(op.selector)) {
+          let timeout = op.timeout || this.defaultTimeout;
+          if (timeout) {
+            await waitUntilElementIsAvailable(this.page, op.selector, timeout);
+          }
           var ret = await _getElementAndBrowserScreenRect(
-            this.page, op.selector
+            this.page, op.selector, op.timeout || 10000
```
Comment (stoarca) [inline src/xdotoolify.js]: is this meant to be 10000 or defaultTimeout?

Comment (stoarca) [reply src/xdotoolify.js]: or rather, should this line not accept a timeout at all since it waits just before?

File: src/xdotoolify.js [hunk 3]
```diff
@@ -298,10 +331,17 @@ _Xdotoolify.prototype.do = async function() {
         commandArr = [];
         await _sleep(op.ms);
       } else if (op.type === 'mousemove') {
+        await this._do(commandArr.join(' '));
```
Comment (stoarca) [inline src/xdotoolify.js]: Not quite understanding why this is done up here.

File: src/xdotoolify.js [hunk 4]
```diff
@@ -377,7 +417,9 @@ _Xdotoolify.prototype.do = async function() {
       }
     }
     if (commandArr.length) {
-      await this._do(commandArr.join(' '));
+      for(let i = 0; i < commandArr.length; i ++) {
```
Comment (stoarca) [inline src/xdotoolify.js]: why did this need to be changed?

File: src/xdotoolify.js [hunk 5]
```diff
@@ -87,21 +91,48 @@ var _getElementAndBrowserRect = async function(page, selector) {
...
+      var elem = null;
+      if (Array.isArray(_selector)) {
+        elem = document.querySelectorAll(_selector[0]);
+      }
+      else {
+        elem = document.querySelector(_selector);
+      }
+      return elem;
+    }, selector);
+    if (element != null) break;
```
Comment (stoarca) [inline src/xdotoolify.js]: please use full braces even for one-line ifs

also use triple equal vs double equal so you get strict comparison
```
if (element !== null) {
  break;
}
```

File: src/xdotoolify.js [hunk 6]
```diff
@@ -63,6 +63,10 @@ var _getElementAndBrowserRect = async function(page, selector) {
         var element = document.querySelector(_selector);
         var result = getElementVisibleBoundingRect(element);
       }
+
+      if (element == null || type(element) == NodeList && element.length == 0) {
```
Comment (stoarca) [inline src/xdotoolify.js]: shouldn't this throw instead of returning?

File: package.json [hunk 7]
```diff
@@ -14,8 +14,7 @@
   },
   "author": "Sergiu Toarca",
   "license": "MIT",
-  "dependencies": {
-  },
+  "dependencies": {},
```
Comment (stoarca) [inline package.json]: plz add version bump as well

File: src/xdotoolify.js (hunk 2 above)
Comment (se468) [reply src/xdotoolify.js]: Good catch

File: src/xdotoolify.js (hunk 3 above)
Comment (se468) [reply src/xdotoolify.js]: It is to wait until everything up to mousemove(which requires ensuring the element with selector exists) is completed. Waiting until the specified timeout happens after that.

**Pull request merged**